### PR TITLE
Moved some validations to the serializer

### PR DESCRIPTION
**Pull request by marianoeramirez**
I moved some of the validations to the serializer.

### CODE REVIEW

File: django_rest_passwordreset/views.py [hunk 1]
```diff
@@ -160,7 +136,7 @@ def post(self, request, *args, **kwargs):
         if not active_user_found and not getattr(settings, 'DJANGO_REST_PASSWORDRESET_NO_INFORMATION_LEAKAGE', False):
             raise exceptions.ValidationError({
                 'email': [_(
-                    "There is no active user associated with this e-mail address or the password can not be changed")],
+                    "We couldn't find an account associate with that email. Please enter another email account.")],
```
Comment (anx-cbenke) [inline django_rest_passwordreset/views.py:139]: Should be `"We couldn't find an account associated with that email. Please try a different e-mail address."`

File: django_rest_passwordreset/serializers.py [hunk 2]
```diff
@@ -10,13 +20,52 @@
...
+        phone_number = to_python(value)
+        if phone_number and phone_number.is_valid():
+            return value
+
+        try:
+            validator = EmailValidator()
+            validator(value)
+            return value
+        except ValidationError:
+            raise ValidationError(_('Enter a valid phone number or email address.'))
```
Comment (anx-cbenke) [inline django_rest_passwordreset/serializers.py:35]: Conflicts with #93, which implements a universal lookup field (Not limited to phone/email)

File: django_rest_passwordreset/views.py [hunk 3]
```diff
@@ -171,19 +147,21 @@ def post(self, request, *args, **kwargs):
...
 
                 # check if the user already has a token
-                if user.password_reset_tokens.all().count() > 0:
-                    # yes, already has a token, re-use this token
-                    token = user.password_reset_tokens.all()[0]
-                else:
+#                 if user.password_reset_tokens.all().count() > 0:
+#                     # yes, already has a token, re-use this token
+#                     token = user.password_reset_tokens.all()[0]
+#                 else:
```
Comment (anx-cbenke) [inline django_rest_passwordreset/views.py:153]: Please remove obsolete code entirely

File: django_rest_passwordreset/serializers.py [hunk 4]
```diff
@@ -10,13 +20,52 @@
...
+
+class PasswordTokenSerializer(PasswordValidateMixin, serializers.Serializer):
     password = serializers.CharField(label=_("Password"), style={'input_type': 'password'})
     token = serializers.CharField()
 
 
-class TokenSerializer(serializers.Serializer):
+class TokenSerializer(PasswordValidateMixin, serializers.Serializer):
     token = serializers.CharField()
+
```
Comment (anx-cbenke) [inline django_rest_passwordreset/serializers.py:71]: This looks much cleaner than the functionality in the post-method.

**Comment by marianoeramirez** [comment]
I close this because i created a new one, #105

**Pull request closed without merging**